### DISCUSSION TITLE
[Feature] separate video transcoding container

### DISCUSSION THREAD

**Discussion by tlvince** [immich-app/immich#3026] [Feature Request]
### The feature

In the same vein as [hosting machine learning container on a different machine](https://immich.app/blog/2023/06/24/update#hosting-machine-learning-container-on-a-different-machine) (via https://github.com/immich-app/immich/pull/2783), doing the same for video transcoding would be useful for resource-constrained environments.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
You can already run the microservices container on another machine, just make sure it can access the database, library, and redis.

**Reply by PathToLife** [reply]
is there a way to limit the types of work of the microservices container?
I've been looking around on the documentation and source code, and haven't found anything that e2e tests multiple microservice containers

**Reply by Pro100x3mal** [reply]
How to push a microservices container to a remote host? I did this for ML, but there is a corresponding IMMICH_MACHINE_LEARNING_URL line in the env file. What is the correct way to add to the main docker-compose.yml url for the remote microservices container?

**Reply by bo0tzz** [reply]
Microservices isn't reached via a URL. It communicates via redis and needs access to all the other backing services (postgres, redis, the library folder, and typesense/ML)

**Reply by tanmaychimurkar** [reply]
Hi there, is it possible to shed some more light on this? I was also looking for a solution to run video transcoding jobs remotely, but could not understand how my remote machine can access postgres, redis, library folder, which are running on the main machine where Immich is running.

**Reply by bo0tzz** [reply]
> how my remote machine can access postgres, redis, library folder

That's up to you. You would need to open ports, set up network shares, stuff like that.

**Comment by bo0tzz** [comment]
> is there a way to limit the types of work of the microservices container?

Not at the moment, though that might be worth adding in the future.

**Comment by zeqiu-mirage** [comment]
> You can already run the microservices container on another machine, just make sure it can access the database, library, and redis.

Is there a guide how to config this? I'm really struggling to let Raspberry Pi to process tons of photos.

**Reply by tanmaychimurkar** [reply]
Hi @zeqiu-mirage, 

There is a similar discussion going on #6989, and I have written a follow-up of how I was able to do this in the comment [here](https://github.com/immich-app/immich/discussions/6989#discussioncomment-10668319). I am also using the same setup with Raspberry Pi hosting the `Immich` instance. I hope this helps in setting up remote microservices jobs.

**Comment by durgeksh** [comment]
I am using the pi 5 8GB as server and other PC (with GPU) as ML service provider by using nginx file as below. 

192.168.1.2 = Pi 5
192.168.1.3 = PC

```
upstream backend {
    server 192.168.1.3:3003 max_fails=1 fail_timeout=1s;
    server 192.168.1.2:3003 backup;
}
server {
    listen 3004;
    server_name 192.168.1.2;
    location / {
        proxy_pass              http://backend/;
    }
}
```

Adjust the docker-compose.yml file for your other services. Just expose the service port and access in pi.
